anvio/cogs.py

```diff
@@ -198,6 +199,16 @@ def process(self, aa_sequences_file_path=None):
         if self.contigs_db_path:
             utils.is_contigs_db(self.contigs_db_path)
 
+        if aa_sequences_file_path:
+            filesnpaths.is_file_exists(aa_sequences_file_path)
+
+            fasta = u.SequenceSource(aa_sequences_file_path); next(fasta) 
+
+            sequence = fasta.seq
+            utils.is_gene_sequence_clean(sequence, amino_acid=True, must_start_with_met=False)
```

Comment:
Thank you very much for finding these functions in utils and making good use of them! Great investigative work :)

Here we are only checking the first sequence. I wonder if it is good enough scrutiny :p I know the trade-off here is compute time for 1 million sequences, but we did learn the hard way quick&dirty checks often lead to a lot of cryptic messages later when things fail.

Reply:
Don't want to waste downstream computation, checked every sequence.